### PR TITLE
[LS] Add foreach and typegurad  completion items  on nested field access

### LINKED ISSUE #30829: foreach statement not provided for class fields
**Description:**
Consider the following source code. 
```ballerina
public type NodeCredential record {|
    string ip;
    int port;
    string username;
|};

public function main() {
    
}

class BoostrapServer {
    private  NodeCredential[] nodes;

    public function init(any arg) {
        self.nodes = [];
    }

    public function addNode(NodeCredential node) {
        self.nodes.push(node);
    }

    public function removeNode(string ip, int port, string username) {
        self.nodes.f<curosr>
    }

}
```
Currently, only the foreach() funcion is suggested as follows

The expected behavior should be as follows. 

**Steps to reproduce:**
Try above code segments on VSCode

**Affected Versions:**
slbeta1-rc5

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #30829 

#### foreach statement

ezgif com-gif-maker(1)

#### typeguard statement

ezgif com-gif-maker

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ForeachCompletionUtil.java [hunk 1]
```diff
@@ -110,11 +111,16 @@ private ForeachCompletionUtil() {
...
+        switch (expr.expression().kind()) {
+            case FIELD_ACCESS:
+            case FUNCTION_CALL:
+                symbolName = expr.expression().toSourceCode().trim();
+                break;
+            case SIMPLE_NAME_REFERENCE:
+                symbolName = ((SimpleNameReferenceNode) expr.expression()).name().text();
+                break;
+            default:
+                symbolName = "items";
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ForeachCompletionUtil.java:123]: What is the use-case of "items" here?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/TypeGuardCompletionUtil.java [hunk 2]
```diff
@@ -65,12 +65,19 @@ public static LSCompletionItem getTypeGuardDestructedItem(UnionTypeSymbol symbol
...
+            case FUNCTION_CALL:
+                //todo: Trim for argument names as well
+            case FIELD_ACCESS:
+                symbolName = expr.expression().toSourceCode().trim();
+                break;
+            case SIMPLE_NAME_REFERENCE:
+                symbolName = ((SimpleNameReferenceNode) expr.expression()).name().text();
+                break;
+            default:
+                symbolName = "name";
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/TypeGuardCompletionUtil.java:78]: what is the use-case for this?

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/TypeGuardCompletionUtil.java (hunk 2 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/TypeGuardCompletionUtil.java:78]: no particular use case. I'll remove it.

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**